protocol/src/fixed_codec/transaction.rs

```diff
@@ -0,0 +1,170 @@
+use bytes::Bytes;
+
+use crate::fixed_codec::{FixedCodecError, ProtocolFixedCodec};
+use crate::types::{
+    primitive::{Balance, ContractType, Fee, Hash, UserAddress},
+    transaction::{CarryingAsset, RawTransaction, SignedTransaction, TransactionAction},
+};
+use crate::{impl_default_fixed_codec_for, ProtocolResult};
+
+// Impl ProtocolFixedCodec trait for types
+impl_default_fixed_codec_for!(transaction, [RawTransaction, SignedTransaction]);
+
+impl rlp::Encodable for RawTransaction {
+    fn rlp_append(&self, s: &mut rlp::RlpStream) {
```

Comment:
Implementing so many nested `RawTransaction` encode can be confusing for the Toolchain.

Maybe we should flatten it? like this
````rs
s.append(TX_FLAG);
s.append(transfer_receiver);
s.append(transfer_amount);
s.append(fee_asset_id);
.....
````

Reply:
ok. working on it